src/openforms/appointments/api/views.py

```diff
@@ -0,0 +1,165 @@
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK
+from rest_framework.views import APIView
+
+from openforms.appointments.api.serializers import (
+    DateInputSerializer,
+    LocationInputSerializer,
+    LocationSerializer,
+    ProductSerializer,
+    TimeInputSerializer,
+)
+from openforms.appointments.base import AppointmentLocation, AppointmentProduct
+from openforms.appointments.utils import get_client
+from openforms.submissions.api.permissions import AnyActiveSubmissionPermission
+from openforms.utils.api.views import ListMixin
+
+
+@extend_schema(
+    summary=_("List available products"),
+)
+class ProductsListView(ListMixin, APIView):
+    """
+    List all products a user can choose when making an appointment.
+    """
+
+    authentication_classes = ()
+    permission_classes = [AnyActiveSubmissionPermission]
+    serializer_class = ProductSerializer
+
+    def get_objects(self):
+        client = get_client()
+        return client.get_available_products()
+
+
+@extend_schema(
+    summary=_("List available locations for a given product"),
+    parameters=[
+        OpenApiParameter(
+            "product_id",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description=_("Id of the product"),
+            required=True,
+        ),
+    ],
+)
+class LocationsListView(ListMixin, APIView):
+    """
+    List all locations for a given product.
+    """
+
+    authentication_classes = ()
+    permission_classes = [AnyActiveSubmissionPermission]
+    serializer_class = LocationSerializer
+
+    def get_objects(self):
+        serializer = LocationInputSerializer(data=self.request.query_params)
+        serializer.is_valid(raise_exception=True)
+
+        product = AppointmentProduct(
+            identifier=serializer.validated_data["product_id"], code="", name=""
+        )
+
+        client = get_client()
+        return client.get_locations([product])
+
+
+@extend_schema(
+    summary=_("List available dates for a given location and product"),
+    parameters=[
+        OpenApiParameter(
+            "product_id",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description=_("Id of the product"),
+            required=True,
+        ),
+        OpenApiParameter(
+            "location_id",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description=_("Id of the location"),
+            required=True,
+        ),
+    ],
+)
+class DatesListView(APIView):
```

Comment:
> Shouldn't this also have the ListMixin?

Since this is returning a list of dates (as strings) I don't think that using a serializer and ListMixin is necessary.  We _could_ do this so that the response is something like `[{'date': '2021-08-23'}, {'date': '2021-08-24'}, ...]` but I think the current response of `['2021-08-23', '2021-08-24', ...]` makes more sense.

I will update the API schema though.

Same with the TimesListView.